eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,43 +1,113 @@
 param (
   $TargetDirectory, # should be in relative form from root of repo. EG: sdk/servicebus
   $RootDirectory, # ideally $(Build.SourcesDirectory)
-  $VsoVariable = "" # target devops output variable
+  $AuthToken,
+  $VsoOwningUsers = "",
+  $VsoOwningTeams = "",
+  $VsoOwningLabels = ""
 )
+
+. (Join-Path ${PSScriptRoot} logging.ps1)
+
 $target = $TargetDirectory.ToLower().Trim("/")
 $codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
 $ownedFolders = @{}
 
 if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
+  LogWarning "Unable to find CODEOWNERS file in target directory $RootDirectory"
   exit 1
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+function VerifyAlias($APIUrl)
+{
+  if ($AuthToken) 
+  {
+    $headers = @{
+      Authorization = "bearer $AuthToken"
+    }
+  }
+  try
+  {
+    $response = Invoke-RestMethod -Headers $headers $APIUrl
+  }
+  catch 
+  {
+    LogDebug "Invoke-RestMethod ${APIUrl} failed with exception:`n$_"
+    LogDebug "This might be because a team alias was used for user API request or vice versa."
+    return $false
   }
+  return $true
 }
 
-$results = $ownedFolders[$target]
+nuget install "CodeOwnersParser" -OutputDirectory . `
+  -Source "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json" `
+  -Prerelease -DirectDownload
 
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
-  
-  if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
+$codeOwnersParserDll = (Get-ChildItem -Path "CodeOwnersParser*" -Filter "CodeOwnersParser.dll" -Recurse).FullName
```

Comment:
I'm not sure how much I like loading this as a dll as that causes other potential issues. If we want to share the logic in the script perhaps it would be better to make a .NET exe we can call that will do the work.

Reply:
I agree, and if we're already pulling in a C# project that handles 80% of the logic, I think we should just move ALL the logic of get-codeowners.ps1 into that program and return a json blob or something that we use to log values to devops variables.

Reply:
In that case, can we add another function to the library for this logic, or some sort of exe wrapper as Wes suggested?

Reply:
Yes, am going the exe wrapper direction, as it turns out that Java is still using it. Although I don't know if that will also be  retired soon.